File: db4oj.tests/src/com/db4o/db4ounit/common/foundation/Arrays4TestCase.java
/* Copyright (C) 2006 Versant Inc. http://www.db4o.com */

package com.db4o.db4ounit.common.foundation;

import com.db4o.foundation.*;

import db4ounit.*;

/**
 * @exclude
 */
public class Arrays4TestCase implements TestCase {

	public void testContainsInstanceOf() {
		Object[] array = new Object[] { "foo", new Integer(42) };
		Assert.isTrue(Arrays4.containsInstanceOf(array, String.class));
		Assert.isTrue(Arrays4.containsInstanceOf(array, Integer.class));
		Assert.isTrue(Arrays4.containsInstanceOf(array, Object.class));
		Assert.isFalse(Arrays4.containsInstanceOf(array, Float.class));
		
		Assert.isFalse(Arrays4.containsInstanceOf(new Object[0], Object.class));
		Assert.isFalse(Arrays4.containsInstanceOf(new Object[1], Object.class));
		Assert.isFalse(Arrays4.containsInstanceOf(null, Object.class));
	}
	
	public void testCopyOfInt() {
		assertCopyOf();
		assertCopyOf(42);
		assertCopyOf(42, 42);
		
		assertCopyOf(new int[] { 1, 2, 3 }, 2);
		assertCopyOf(new int[] { 1, 2 }, 3);
	}

	private void assertCopyOf(int[] array, int newLength) {
		assertCopyOf(expectationFor(array, newLength), array, newLength);
	}

	private int[] expectationFor(int[] array, int newLength) {
		final int[] expectation = new int[newLength];
		System.arraycopy(array, 0, expectation, 0, Math.min(array.length, newLength));
		return expectation;
	}

	private void assertCopyOf(final int... array) {
		assertCopyOf(array, array, array.length);
	}

	private void assertCopyOf(final int[] expected, final int[] array, final int newLength) {
		final int[] copy = Arrays4.copyOf(array, newLength);
		Assert.areNotSame(array, copy);
		ArrayAssert.areEqual(expected, copy);
	}

}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Foundation/Arrays4TestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4oUnit;
using Db4objects.Db4o.Foundation;
using Sharpen;

namespace Db4objects.Db4o.Tests.Common.Foundation
{
	/// <exclude></exclude>
	public class Arrays4TestCase : ITestCase
	{
		public virtual void TestContainsInstanceOf()
		{
			object[] array = new object[] { "foo", 42 };
			Assert.IsTrue(Arrays4.ContainsInstanceOf(array, typeof(string)));
			Assert.IsTrue(Arrays4.ContainsInstanceOf(array, typeof(int)));
			Assert.IsTrue(Arrays4.ContainsInstanceOf(array, typeof(object)));
			Assert.IsFalse(Arrays4.ContainsInstanceOf(array, typeof(float)));
			Assert.IsFalse(Arrays4.ContainsInstanceOf(new object[0], typeof(object)));
			Assert.IsFalse(Arrays4.ContainsInstanceOf(new object[1], typeof(object)));
			Assert.IsFalse(Arrays4.ContainsInstanceOf(null, typeof(object)));
		}

		public virtual void TestCopyOfInt()
		{
			AssertCopyOf(new int[] {  });
			AssertCopyOf(new int[] { 42 });
			AssertCopyOf(new int[] { 42, 42 });
			AssertCopyOf(new int[] { 1, 2, 3 }, 2);
			AssertCopyOf(new int[] { 1, 2 }, 3);
		}

		private void AssertCopyOf(int[] array, int newLength)
		{
			AssertCopyOf(ExpectationFor(array, newLength), array, newLength);
		}

		private int[] ExpectationFor(int[] array, int newLength)
		{
			int[] expectation = new int[newLength];
			System.Array.Copy(array, 0, expectation, 0, Math.Min(array.Length, newLength));
			return expectation;
		}

		private void AssertCopyOf(int[] array)
		{
			AssertCopyOf(array, array, array.Length);
		}

		private void AssertCopyOf(int[] expected, int[] array, int newLength)
		{
			int[] copy = Arrays4.CopyOf(array, newLength);
			Assert.AreNotSame(array, copy);
			ArrayAssert.AreEqual(expected, copy);
		}
	}
}
